### PR TITLE
feat: [P4PU-404] added updated filter for logout

### PR DESCRIPTION
**Pull request by oleksiybozhykntt**
#### List of Changes
added filter for logout action
added unit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/java/it/gov/pagopa/arc/security/CustomLogoutHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Mock
+  private Authentication authentication;
+
+  @InjectMocks
+  private CustomLogoutHandler customLogoutHandler;
+
+  @Test
+  void givenLogoutRequestThenPerformLogout() {
+    MockitoAnnotations.openMocks(this);
```
Comment (and-mora) [inline src/test/java/it/gov/pagopa/arc/security/CustomLogoutHandlerTest.java:37]: use class annotation `@ExtendWith(MockitoExtension.class)` instead

Comment (oleksiybozhykntt) [reply src/test/java/it/gov/pagopa/arc/security/CustomLogoutHandlerTest.java:37]: done

File: src/test/java/it/gov/pagopa/arc/security/CustomLogoutHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    String authorizationHeader = "Bearer " + token;
+
+    when(request.getHeader(HttpHeaders.AUTHORIZATION)).thenReturn(authorizationHeader);
+    when(tokenStoreService.delete(token)).thenReturn(Optional.empty());
+
+    customLogoutHandler.logout(request, response, authentication);
+
+    verify(tokenStoreService, times(1)).delete(token);
+    verify(request, times(1)).getHeader(HttpHeaders.AUTHORIZATION);
+    verify(tokenStoreService,times(1)).delete(token);
```
[the remark this answers is not in the document]
Comment (oleksiybozhykntt) [reply src/test/java/it/gov/pagopa/arc/security/CustomLogoutHandlerTest.java:49]: yes removed

**Comment by and-mora** [comment]
there is a branch condition uncovered (see sonar), @oleksiybozhykntt can you try to cover it?

**Review by and-mora — APPROVED** [review]

**Review by and-mora — APPROVED** [review]

**Pull request merged**